From 373bdce3f80f4c1a468e0f33e6d78548b377aadc Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@denx.de>
Date: Mon, 25 Jul 2022 14:21:17 +0200
Subject: [PATCH 61/61] arm64: dts: imx8mp: Adjust DT to match downstream DTSI

The downstream kernel fork DTSI contain various differently named nodes
and custom properties, adjust the mainline DT to match the downstream
DTSI.

Upstream-Status: Inappropriate
Signed-off-by: Marek Vasut <marex@denx.de>
---
 ...mx8mp-dhcom-overlay-panel-ch101olhlwh.dtsi |   8 -
 .../imx8mp-dhcom-overlay-panel-lvds.dtsi      |  20 ++-
 .../imx8mp-dhcom-pdk2-downstream.dtsi         |  14 ++
 .../boot/dts/freescale/imx8mp-dhcom-pdk2.dts  |   2 +-
 .../imx8mp-dhcom-pdkx-downstream.dtsi         | 149 ++++++++++++++++++
 .../imx8mp-dhcom-som-downstream.dtsi          |  25 +++
 .../boot/dts/freescale/imx8mp-dhcom-som.dtsi  |   8 +-
 7 files changed, 211 insertions(+), 15 deletions(-)
 create mode 100644 arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk2-downstream.dtsi
 create mode 100644 arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdkx-downstream.dtsi
 create mode 100644 arch/arm64/boot/dts/freescale/imx8mp-dhcom-som-downstream.dtsi

diff --git a/arch/arm64/boot/dts/freescale/imx8mp-dhcom-overlay-panel-ch101olhlwh.dtsi b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-overlay-panel-ch101olhlwh.dtsi
index 99a923f17b1f3..541eb0ce42a41 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-dhcom-overlay-panel-ch101olhlwh.dtsi
+++ b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-overlay-panel-ch101olhlwh.dtsi
@@ -34,14 +34,6 @@ eeprom@50 {
 	};
 };
 
-&lcdif2 {
-	/*
-	 * The Chefree CH101OLHLWH-002 panel requires 71.4 MHz LVDS clock.
-	 * Set IMX8MP_VIDEO_PLL1 to 500 MHz , since 500 MHz / 7 = 71.42 MHz .
-	 */
-	assigned-clock-rates = <0>, <500000000>;
-};
-
 &panel {
 	compatible = "chefree,ch101olhlwh-002";
 };
diff --git a/arch/arm64/boot/dts/freescale/imx8mp-dhcom-overlay-panel-lvds.dtsi b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-overlay-panel-lvds.dtsi
index 8dc5353e77313..f335e9dd2d563 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-dhcom-overlay-panel-lvds.dtsi
+++ b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-overlay-panel-lvds.dtsi
@@ -9,11 +9,25 @@ &lcdif2 {
 	status = "okay";
 };
 
-&ldb_lvds_ch0 {
-	remote-endpoint = <&panel_in>;
+&ldb {
+	status = "okay";
+
+	lvds-channel@0 {
+		fsl,data-mapping = "spwg";
+		fsl,data-width = <24>;
+		status = "okay";
+
+		port@1 {
+			reg = <1>;
+
+			ldb_lvds_ch0: endpoint {
+				remote-endpoint = <&panel_in>;
+			};
+		};
+	};
 };
 
-&lvds_bridge {
+&ldb_phy {
 	status = "okay";
 };
 
diff --git a/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk2-downstream.dtsi b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk2-downstream.dtsi
new file mode 100644
index 0000000000000..dbeb05bcc12c2
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk2-downstream.dtsi
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (C) 2023 Marek Vasut <marex@denx.de>
+ */
+
+#include "imx8mp-dhcom-pdkx-downstream.dtsi"
+
+&pcie_phy {
+	ext_osc = <0>;
+};
+
+&pcie {
+	ext_osc = <0>;
+};
diff --git a/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk2.dts b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk2.dts
index 937be7df8a8a2..83843cdd3af26 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk2.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk2.dts
@@ -11,8 +11,8 @@
 /dts-v1/;
 
 #include <dt-bindings/leds/common.h>
-#include <dt-bindings/phy/phy-imx8-pcie.h>
 #include "imx8mp-dhcom-som.dtsi"
+#include "imx8mp-dhcom-pdk2-downstream.dtsi"
 
 / {
 	model = "DH electronics i.MX8M Plus DHCOM Premium Developer Kit (2)";
diff --git a/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdkx-downstream.dtsi b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdkx-downstream.dtsi
new file mode 100644
index 0000000000000..7796b6d0d17bf
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdkx-downstream.dtsi
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (C) 2023 Marek Vasut <marex@denx.de>
+ */
+
+#define IMX8_PCIE_REFCLK_PAD_UNUSED	0
+#define IMX8_PCIE_REFCLK_PAD_INPUT	1
+
+/ {
+	sound-hdmi {
+		compatible = "fsl,imx-audio-hdmi";
+		model = "audio-hdmi";
+		audio-cpu = <&aud2htx>;
+		hdmi-out;
+		constraint-rate = <44100>,
+				<88200>,
+				<176400>,
+				<32000>,
+				<48000>,
+				<96000>,
+				<192000>;
+		status = "okay";
+	};
+};
+
+&aips5 {
+	hdmi_tx: hdmi@32fd8000 { };
+	hdmi_tx_phy: hdmiphy@32fdff00 { };
+};
+
+&aud2htx {
+	status = "okay";
+};
+
+&clk {
+	assigned-clocks = <&clk IMX8MP_CLK_A53_SRC>,
+			  <&clk IMX8MP_CLK_A53_CORE>,
+			  <&clk IMX8MP_CLK_NOC>,
+			  <&clk IMX8MP_CLK_NOC_IO>,
+			  <&clk IMX8MP_CLK_GIC>,
+			  <&clk IMX8MP_CLK_AUDIO_AHB>,
+			  <&clk IMX8MP_CLK_AUDIO_AXI_SRC>,
+			  <&clk IMX8MP_CLK_MEDIA_DISP1_PIX>,
+			  <&clk IMX8MP_CLK_MEDIA_AXI>,
+			  <&clk IMX8MP_CLK_MEDIA_APB>,
+			  <&clk IMX8MP_AUDIO_PLL1>,
+			  <&clk IMX8MP_VIDEO_PLL1>;
+	assigned-clock-parents = <&clk IMX8MP_SYS_PLL1_800M>,
+				 <&clk IMX8MP_ARM_PLL_OUT>,
+				 <&clk IMX8MP_SYS_PLL2_1000M>,
+				 <&clk IMX8MP_SYS_PLL1_800M>,
+				 <&clk IMX8MP_SYS_PLL2_500M>,
+				 <&clk IMX8MP_SYS_PLL1_800M>,
+				 <&clk IMX8MP_SYS_PLL1_800M>,
+				 <&clk IMX8MP_VIDEO_PLL1_OUT>,
+				 <&clk IMX8MP_SYS_PLL1_800M>,
+				 <&clk IMX8MP_SYS_PLL1_800M>;
+	assigned-clock-rates = <0>, <0>,
+			       <1000000000>,
+			       <800000000>,
+			       <500000000>,
+			       <400000000>,
+			       <800000000>,
+			       <0>,
+			       <400000000>,
+			       <200000000>,
+			       <393216000>,
+			       <1039500000>;
+};
+
+&dsp {
+	status = "okay";
+};
+
+&irqsteer_hdmi {
+	status = "okay";
+};
+
+&hdmi_blk_ctrl {
+	status = "okay";
+};
+
+&hdmi_pavi {
+	status = "okay";
+};
+
+&lcdif3 {
+	status = "okay";
+
+	thres-low  = <1 2>;             /* (FIFO * 1 / 2) */
+	thres-high = <3 4>;             /* (FIFO * 3 / 4) */
+};
+
+&pcie_phy {
+	/delete-property/ clock-names;
+	/delete-property/ clocks;
+	/delete-property/ fsl,clkreq-unsupported;
+	/delete-property/ fsl,refclk-pad-mode;
+};
+
+&pcie {
+	clocks = <&clk IMX8MP_CLK_HSIO_ROOT>,
+		 <&clk IMX8MP_CLK_PCIE_AUX>,
+		 <&clk IMX8MP_CLK_HSIO_AXI>,
+		 <&clk IMX8MP_CLK_PCIE_ROOT>;
+	clock-names = "pcie", "pcie_aux", "pcie_phy", "pcie_bus";
+	assigned-clocks = <&clk IMX8MP_CLK_HSIO_AXI>,
+			  <&clk IMX8MP_CLK_PCIE_AUX>;
+	assigned-clock-rates = <500000000>, <10000000>;
+	assigned-clock-parents = <&clk IMX8MP_SYS_PLL2_500M>,
+				 <&clk IMX8MP_SYS_PLL2_50M>;
+	l1ss-disabled;
+};
+
+&sdma2 {
+	status = "okay";
+};
+
+&vpu_g1 {
+	status = "okay";
+};
+
+&vpu_g2 {
+	status = "okay";
+};
+
+&vpu_vc8000e {
+	status = "okay";
+};
+
+&vpu_v4l2 {
+	status = "okay";
+};
+
+&gpu_3d {
+	status = "okay";
+};
+
+&gpu_2d {
+	status = "okay";
+};
+
+&ml_vipsi {
+	status = "okay";
+};
+
+&mix_gpu_ml {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/freescale/imx8mp-dhcom-som-downstream.dtsi b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-som-downstream.dtsi
new file mode 100644
index 0000000000000..e4d046a2879c5
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-som-downstream.dtsi
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (C) 2023 Marek Vasut <marex@denx.de>
+ */
+
+#define MX8MP_IOMUXC_GPIO1_IO10__USB1_OTG_ID	MX8MP_IOMUXC_GPIO1_IO10__USB1_ID
+#define MX8MP_IOMUXC_GPIO1_IO14__USB2_OTG_PWR	MX8MP_IOMUXC_GPIO1_IO14__USB2_PWR
+#define MX8MP_IOMUXC_GPIO1_IO15__USB2_OTG_OC	MX8MP_IOMUXC_GPIO1_IO15__USB2_OC
+
+&mipi_dsi {
+	samsung,burst-clock-frequency = <160000000>;
+	samsung,esc-clock-frequency = <10000000>;
+
+	/delete-node/ port@0;
+	ports {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		port@0 {
+			dsim_from_lcdif: endpoint {
+				remote-endpoint = <&lcdif_to_dsim>;
+			};
+		};
+	};
+};
diff --git a/arch/arm64/boot/dts/freescale/imx8mp-dhcom-som.dtsi b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-som.dtsi
index a03a29b0d510f..1ae78ba5f1956 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-dhcom-som.dtsi
+++ b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-som.dtsi
@@ -4,6 +4,7 @@
  */
 
 #include "imx8mp.dtsi"
+#include "imx8mp-dhcom-som-downstream.dtsi"
 
 / {
 	model = "DH electronics i.MX8M Plus DHCOM SoM";
@@ -440,6 +441,7 @@ port@1 {
 			dsi_out: endpoint {
 				data-lanes = <1 2 3 4>;
 				remote-endpoint = <&tc_bridge_in>;
+				attach-bridge;
 			};
 		};
 	};
@@ -939,14 +941,14 @@ MX8MP_IOMUXC_UART4_TXD__UART4_DCE_TX		0x49
 
 	pinctrl_usb0_vbus: dhcom-usb0-grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_GPIO1_IO10__USB1_OTG_ID		0x0
+			MX8MP_IOMUXC_GPIO1_IO10__USB1_ID		0x0
 		>;
 	};
 
 	pinctrl_usb1_vbus: dhcom-usb1-grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_GPIO1_IO14__USB2_OTG_PWR		0x6
-			MX8MP_IOMUXC_GPIO1_IO15__USB2_OTG_OC		0x80
+			MX8MP_IOMUXC_GPIO1_IO14__USB2_PWR		0x6
+			MX8MP_IOMUXC_GPIO1_IO15__USB2_OC		0x80
 		>;
 	};
 
-- 
2.39.1

